vendor/github.com/livepeer/lpms/vidplayer/player.go

```diff
@@ -125,7 +126,11 @@ func handleLive(w http.ResponseWriter, r *http.Request,
 
 		mediaPl, err = getMediaPlaylist(r.URL)
 		if err != nil {
-			http.Error(w, "Error getting HLS media playlist", 500)
+			if err == ErrNotFound {
+				http.Error(w, "ErrNotFound", 404)
+			} else {
+				http.Error(w, "Error getting HLS media playlist", 500)
```

Comment:
What is the scenario when we actually want to return a 500? Some sort of internal inconsistency or error?

Reply:
Up to debate, but I think any "unexpected internal error" should be 500.

Reply:
Typically, `500` means "sorry something went wrong, but we're not sure what exactly". So it would be good to return more informative error codes if we do know why it failed.